### DISCUSSION TITLE
How to implement blockchain on OMNET++

### DISCUSSION THREAD

**Discussion by bbnorth** [omnetpp/omnetpp#1158] [Q&A]
How to implement blockchain on OMNET++

**Comment by rhornig** [comment]
"Blockchain" is a data structure that does not have any time dependent behavior, so you won't need to call any OMNeT++ APIs. So the short answer:

Exactly how you would implement it in C++.

**Reply by bbnorth** [reply]
Thank you very much for your reply. I am new to OMNET++ and hope you can give me more advice.
Do you mean that blockchain can be directly implemented in OMNET++? If RSU serves as the consensus node of the blockchain, such as the PBFT consensus algorithm, can the PBFT consensus algorithm only be written in C++ and written into the RSU module?

**Reply by bbnorth** [reply]
Do you mean that if RSU acts as a consensus node, the consensus algorithm written in C++ needs to be written into the RSU module?

**Comment by bbnorth** [comment]
Are there any reference materials for me to learn how to simulate
blockchain in OMNET++?

Rudolf Hornig ***@***.***> 于2024年5月14日周二 16:48写道：

**Reply by YGmoyu** [reply]
some paper in scholar google，which run vehicles data in omnet and import the data to Hyperledger fabric ,mabey you can try that.

**Reply by YGmoyu** [reply]
you can run this project to learning more about it https://github.com/Adityak9/vanet_blockchain
